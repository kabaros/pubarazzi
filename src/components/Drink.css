
.drink {
    flex: 1;
    height: 100px;
    width: 30%;
    flex-basis: 30%;
    flex-wrap: wrap;
    margin: 5px;
    display: inline-block;
    border: 1px solid #D8D8D8;
    cursor: pointer;
    position: relative;
}

.drink:hover {
    border: 1px solid #2C2C2C;
}

.drink.highlighted {
    background-color: #83E791;
    border: 1px solid #818145;
}
.drink.highlighted-warning {
    background-color: yellow;
}

.drink.highlighted-danger {
    background-color: red;
}

.drink.highlighted:hover {
    border: 1px solid green;
}

.drink h1 {
    font-size: 14px;
    text-align: center;
}

.drink .add-drink {
    position: absolute;
    right: 5px;
    bottom: 0px;
    font-size: 28px;
}

.drink .order-total {
    position: absolute;
    left: 30px;
    bottom: 5px;
    margin: 0;
    font-size: 0.6em;
}