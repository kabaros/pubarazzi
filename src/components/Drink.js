import React from 'react';
import './Drink.css';

export default (props) => {
    const { drink, order, addDrink } = props;
    const highlighted = order.count > 0? 'highlighted': '';

    const warning = order.count > 2 ? (order.count < 6? '-warning': '-danger') : '';
    
    return (
        <div className={`drink ${highlighted}${warning}`}>
            <h1>{drink.name}</h1>
            <h2>{order.count}</h2>
            
            <div onClick={() => addDrink(drink)} 
                className='add-drink'>+</div>
        </div>
    )
}