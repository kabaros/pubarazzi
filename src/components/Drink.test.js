import React from 'react';
import { render } from 'react-dom';
import Drink from './Drink';

it('renders a Drink component', () => {
    const div = document.createElement('div');
    const drink = {
        name: 'mojito'
    };
    const order = {};

    render(<Drink drink={drink} order={order} />, div);

    expect(div).toMatchSnapshot();
});


it('renders a Drink component with a red background if more than 5 orders', () => {
    const div = document.createElement('div');
    const drink = {
        id: '1',
        name: 'mojito'
    };
    const order = {
        count: 10
    };

    render(<Drink drink={drink} order={order} />, div);

    expect(div.innerHTML).toMatch('highlighted-danger')
});


it('renders a Drink component with a warning background if more than 2 and less than 5 5 orders', () => {
    const div = document.createElement('div');
    const drink = {
        id: '1',
        name: 'mojito'
    };
    const order = {
        count: 3
    };

    render(<Drink drink={drink} order={order} />, div);

    expect(div.innerHTML).toMatch('highlighted-warning')
});

