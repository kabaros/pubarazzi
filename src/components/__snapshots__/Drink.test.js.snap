// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Drink component 1`] = `
<div>
  <div
    class="drink "
  >
    <h1>
      mojito
    </h1>
    <h2 />
    <div
      class="add-drink"
    >
      +
    </div>
  </div>
</div>
`;
